lib/ui/painting.dart

```diff
@@ -307,6 +307,15 @@ class Color {
     }
   }
 
+  /// Returns an alpha value representative of the provided [opacity] value.
+  ///
+  /// The [opacity] value may not be null.
+  static int getAlphaFromOpacity(double opacity) {
+    assert(opacity != null);
+    final double clampedOpacity = opacity.clamp(0.0, 1.0);
+    return (clampedOpacity * 255).round();
```

Comment:
You could probably combine this all into one line:
```suggestion
    return (opacity.clamp(0.0, 1.0) * 255).round();
```